src/vivarium/framework/results/observation.py

```diff
@@ -313,22 +317,22 @@ def get_complete_stratified_results(
 
     @staticmethod
     def _aggregate(
-        pop_groups: DataFrameGroupBy,
-        aggregator_sources: Optional[list[str]],
-        aggregator: Callable[[pd.DataFrame], Union[float, pd.Series]],
-    ) -> Union[pd.Series, pd.DataFrame]:
+        pop_groups: DataFrameGroupBy[Any],
+        aggregator_sources: list[str] | None,
+        aggregator: Callable[[pd.DataFrame], float | pd.Series[float]],
+    ) -> pd.Series[float] | pd.DataFrame:
         """Apply the `aggregator` to the population groups and their
         `aggregator_sources` columns.
         """
         aggregates = (
-            pop_groups[aggregator_sources].apply(aggregator).fillna(0.0)
+            pop_groups[aggregator_sources].apply(aggregator).fillna(0.0)  # type: ignore [arg-type]
             if aggregator_sources
-            else pop_groups.apply(aggregator)
+            else pop_groups.apply(aggregator)  # type: ignore [arg-type]
```

Comment:
These two ignores are cause by the DataFrameGroupBy objects needing a parameter type hint. In reality, it isn't any but it is a lot of things and so I am using any. I also chose to ignore it here so the typing for the function is a little more help to users instead of having an arg and the return type be Any.

Reply:
@rmudambi Is it reasonable to type hint the DataFrameGroupBys as `DataFrameGroupBy[str]`? I'm trying to think of the types of things we stratify by and they're all strings (age group, sex, location, disease states, etc). But I guess we might one day want to stratify by other things?

Or maybe `DataFrameGroupBy[str, int]` just to allow for IDs?